### PR TITLE
Reverse additional edges

### PR DESCRIPTION
**Pull request by halseth**
Make additional edges passed to `findPath` incoming, so we don't have to reverse them.

### CODE REVIEW

**Comment by joostjager** [comment]
I think we discussed this idea also in the backwards search PR. The reason not to make the change then was that the parameter `additionalEdges map[Vertex][]*edgePolicyWithSource` contains redundant information, as the source is both the map key and a field of `edgePolicyWithSource`. We didn't find it good to expose this redundancy to callers of the function as it is an implementation detail.

**Comment by halseth** [comment]
Yeah, I remember. However, I think this is the way to go, as it reduces overall code complexity, and ties in nicely with an upcoming change where I hide all edges (normal+additional) behind a common interface. We can probably hold off proper review on this until that PR is up, for the bigger picture.

File: routing/missioncontrol.go [hunk 1]
```diff
@@ -215,8 +215,14 @@ func (m *missionControl) NewPaymentSession(routeHints [][]HopHint,
...
 			}
 
-			v := NewVertex(hopHint.NodeID)
-			edges[v] = append(edges[v], edge)
+			from := NewVertex(hopHint.NodeID)
+			to := Vertex(endNode.PubKeyBytes)
+
+			fromNode := &channeldb.LightningNode{PubKeyBytes: from}
+			edges[to] = append(edges[to], &edgePolicyWithSource{
+				sourceNode: fromNode,
```
Comment (cfromknecht) [inline routing/missioncontrol.go]: Might it be worth making `soureNode` a `[33]byte` value directly? We waste a lot of space allocating `channeldb.LightningNode`, to only use one field

First commit message appears to be clipped

**Comment by halseth** [comment]
Setting PR `blocked` and making `P4`, as this might be incorporated or become obsolete after the planned Mission Control changes.

File: routing/missioncontrol.go (hunk 1 above)
Comment (cfromknecht) [reply routing/missioncontrol.go]: any updates on this?

**Comment by cfromknecht** [comment]
Ah nevermind, now seeing your last comment

**Pull request closed without merging**